score-vmware/src/main/java/io/cloudslang/content/actions/vm/CreateVM.java

```diff
@@ -0,0 +1,94 @@
+
+    @Action(name = "Create Virtual Machine",
+            outputs = {
+                    @Output(Outputs.RETURN_CODE),
+                    @Output(Outputs.RETURN_RESULT),
+                    @Output(Outputs.EXCEPTION)
+            },
+            responses = {
+                    @Response(text = Outputs.SUCCESS, field = Outputs.RETURN_CODE, value = Outputs.RETURN_CODE_SUCCESS,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.RESOLVED),
+                    @Response(text = Outputs.FAILURE, field = Outputs.RETURN_CODE, value = Outputs.RETURN_CODE_FAILURE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.ERROR, isOnFail = true)
+            })
+    public Map<String, String> createVM(@Param(value = Inputs.HOST, required = true) String host,
+                                        @Param(Inputs.PORT) String port,
+                                        @Param(Inputs.PROTOCOL) String protocol,
+                                        @Param(Inputs.USERNAME) String username,
+                                        @Param(value = Inputs.PASSWORD, encrypted = true) String password,
+                                        @Param(Inputs.X509_HOSTNAME_VERIFIER) String x509HostnameVerifier,
+
+                                        @Param(value = Inputs.DATA_CENTER_NAME, required = true) String dataCenterName,
+                                        @Param(Inputs.HOSTNAME) String hostname,
+                                        @Param(value = Inputs.VIRTUAL_MACHINE_NAME, required = true) String virtualMachineName,
+                                        @Param(Inputs.VIRTUAL_MACHINE_DESCRIPTION) String description,
+                                        @Param(value = Inputs.DATA_STORE, required = true) String dataStore,
+                                        @Param(Constants.RESOURCE_POOL) String resourcePool,
+                                        @Param(Inputs.CPU_COUNT) String numCPUs,
+                                        @Param(Inputs.VM_DISK_SIZE) String vmDiskSize,
+                                        @Param(Inputs.VM_MEMORY_SIZE) String vmMemorySize,
+                                        @Param(value = Inputs.GUEST_OS_ID, required = true) String guestOsId) {
+
+        Map<String, String> resultMap = new HashMap<>();
+
+        int intPort = InputUtils.getIntInput(port, Inputs.DEFAULT_HTTPS_PORT);
+        boolean booleanX509HostnameVerifier = InputUtils.getBooleanInput(x509HostnameVerifier);
+        int intNumCPUs = InputUtils.getIntInput(numCPUs, Constants.DEFAULT_CPU_COUNT);
+        long longVmDiskSize = InputUtils.getLongInput(vmDiskSize, Constants.DEFAULT_VM_DISK_SIZE_MB);
+        long longVmMemorySize = InputUtils.getLongInput(vmMemorySize, Constants.DEFAULT_VM_MEMORY_SIZE_MB);
+
+        try {
+
+            ManagedObjectReference managedObjectReference = service.createVirtualMachine(host,
```

Comment:
the service could return the resultMap, we should not know about the existence of ManagedObjectReference at this point
also the createVirtualMachine method has too many parameters